cli/src/main/java/org/jboss/as/cli/handlers/ShutdownHandler.java

```diff
@@ -244,6 +298,45 @@ protected boolean isLocalHost(ModelControllerClient client, String host) throws
         return result.asString().equals(host);
     }
 
+    protected boolean isLocalClientLaunch(CommandContext ctx) throws CommandLineException {
+        final ModelNode op = new ModelNode();
+        final ParsedCommandLine args = ctx.getParsedCommandLine();
+        final ModelControllerClient client = ctx.getModelControllerClient();
+        if(ctx.isDomainMode()) {
+            final String hostName = host.getValue(args);
+            if (hostName == null) {
+                throw new CommandFormatException("Missing required argument " + host.getFullName());
+            }
+            op.get(Util.ADDRESS).add(Util.HOST, hostName);
+            op.get(Util.ADDRESS).add(Util.CORE_SERVICE, "host-environment");
+
+        } else {
+            op.get(Util.ADDRESS).add(Util.CORE_SERVICE, "server-environment");
+        }
+        op.get(Util.OPERATION).set("query");
```

Comment:
@yersan Using query is more data. (Note I said read-attribute, not read-resource.)

```
[standalone@embedded /] /core-service=server-environment:read-attribute(name=home-dir)
{
    "outcome" => "success",
    "result" => "/Users/bstansberry/dev/wildfly/wildfly-core/build/target/wildfly-core-21.0.0.Beta1-SNAPSHOT"
}

[standalone@embedded /] /core-service=server-environment:query(select=[home-dir])
{
    "outcome" => "success",
    "result" => {"home-dir" => "/Users/bstansberry/dev/wildfly/wildfly-core/build/target/wildfly-core-21.0.0.Beta1-SNAPSHOT"}
}
```

Using 'query' is why on L336 you have to do 'result.get(Util.HOME_DIR).asString()' instead of simply 'result.asString()'.